bindings/profilers/heap.cc

```diff
@@ -85,18 +304,78 @@ NAN_METHOD(HeapProfiler::StartSamplingHeapProfiler) {
 // Signature:
 // stopSamplingHeapProfiler()
 NAN_METHOD(HeapProfiler::StopSamplingHeapProfiler) {
-  info.GetIsolate()->GetHeapProfiler()->StopSamplingHeapProfiler();
+  auto isolate = info.GetIsolate();
+  isolate->GetHeapProfiler()->StopSamplingHeapProfiler();
+  auto state = PerIsolateData::For(isolate)->GetHeapProfilerState();
+  if (state) {
+    isolate->RemoveNearHeapLimitCallback(&NearHeapLimit, 0);
```

Comment:
Not sure if it's intentional, but this _does_ mean that if you stop and restart the heap profiler you will need to call the MonitorOutOfMemory function again too.

Reply:
I am not sure what would be the least surprising behaviour from the client perspective here.
Currently `stop` acts as a full reset to the state prior to `start`.